cylc/flow/option_parsers.py

```diff
@@ -290,9 +290,15 @@ class CylcOptionParser(OptionParser):
             ['--debug'], help='Equivalent to -v -v',
             dest='verbosity', action='store_const', const=2, useif='all'),
         OptionSettings(
-            ['--no-timestamp'], help='Don\'t timestamp logged messages.',
-            action='store_false', dest='log_timestamp',
-            default=True, useif='all'),
+            ['--timestamp'],
+            help='Add a timestamp to messages logged to the terminal.',
+            action='store_true', dest='log_timestamp',
+            default=False, useif='all'),
+        OptionSettings(
+            ['--no-timestamp'], help="Don't add a timestamp to messages logged"
+            " to the terminal (this does nothing - it is now the default.",
+            action='store_false', dest='_noop',
+            default=False, useif='all'),
```

Comment:
Off-topic?

Reply:
Arguably yes, but it was motivated by the fact that this PR touches code used by both the CLI and the scheduler, so log messages can appear in the terminal, in which case the long timestamp sticks out like a sore thumb and is really not helpful.

This means `cylc play --no-detach` won't log timestamps to the terminal by default, but I don't see any downside to that. The log file itself still has timestamps.

So I'd prefer to leave this in.

Reply:
It just means it will take longer to review...

Reply:
It's a tiny change! (But if you really insist, I'll take it out).

Reply:
Maybe, but these things add up!

Reply:
Thanks (and apologies 😁  - but I don't think there's other off-topic changes in this PR)